pkgs/coverage/lib/src/coverage_options.dart

```diff
@@ -0,0 +1,78 @@
+import 'dart:io';
+import 'package:cli_config/cli_config.dart';
+
+class CoverageOptions {
+  const CoverageOptions({
+    required this.output,
+    required this.scopeOutput,
+    required this.functionCoverage,
+    required this.branchCoverage,
+    required this.packagePath,
+    this.packageName,
+    required this.testScript,
+  });
+
+  factory CoverageOptions.fromConfig(
+      Config options, CoverageOptions defaultOptions) {
+    return CoverageOptions(
+      output: options.optionalString('out') ?? defaultOptions.output,
+      scopeOutput: options.optionalStringList('scope_output') ??
+          defaultOptions.scopeOutput,
+      functionCoverage: options.optionalBool('function_coverage') ??
+          defaultOptions.functionCoverage,
+      branchCoverage: options.optionalBool('branch_coverage') ??
+          defaultOptions.branchCoverage,
+      packagePath:
+          options.optionalString('package') ?? defaultOptions.packagePath,
+      packageName:
+          options.optionalString('package_name') ?? defaultOptions.packageName,
+      testScript: options.optionalString('test') ?? defaultOptions.testScript,
+    );
+  }
+
+  final String? output;
+  final List<String> scopeOutput;
+  final bool functionCoverage;
+  final bool branchCoverage;
+  final String packagePath;
+  final String? packageName;
+  final String testScript;
+}
+
+class CoverageOptionsProvider {
+  CoverageOptionsProvider({
+    String? filePath,
+  }) {
+    final file = _getOptionsFile(filePath ?? CoverageOptionsProvider.filePath);
```

Comment:
This line implicitly evaluates `CoverageOptionsProvider.filePath` relative to the current working directory of the dart script. This means if the user runs the coverage command from a subdirectory of the package, the coverage options won't be picked up.

So change it to search upwards from the current directory until it finds a coverage options file. If it finds a pubspec.yaml file it should quit searching (we don't want to search outside the package dir). If `filePath` is non-null, you don't need to do any searching.